code/game/turfs/space/space.dm

```diff
@@ -15,6 +15,13 @@
 /turf/space/New()
 	if(!istype(src, /turf/space/transit))
 		icon_state = "[((x + y) ^ ~(x * y) + z) % 25]"
+	update_starlight()
+
+/turf/space/proc/update_starlight()
+	for(var/turf/T in range(src,1))
+		if(istype(T,/turf/simulated))
+			src.SetLuminosity(2)
+			return
 

```

Comment:
update_starlight() has to be called when constructing and deconstructing turfs, to keep the luminosity updated properly.
After the loop, add the line 'src.SetLuminosity(0)' or whatever, to remove luminosity, if necessary.